api/admin/routes.py

```diff
@@ -32,19 +32,20 @@
 import urllib
 from datetime import timedelta
 
-# The secret key is used for signing cookies for admin login
-app.secret_key = ConfigurationSetting.sitewide_secret(
-    _db, Configuration.SECRET_KEY
-)
-
 # An admin's session will expire after this amount of time and
 # the admin will have to log in again.
 app.permanent_session_lifetime = timedelta(hours=9)
 
 @app.before_first_request
-def setup_admin():
+def setup_admin(_db=None):
     if getattr(app, 'manager', None) is not None:
         setup_admin_controllers(app.manager)
+    _db = _db or app._db
+    # The secret key is used for signing cookies for admin login
+    app.secret_key = ConfigurationSetting.sitewide_secret(
+        _db, Configuration.SECRET_KEY
+    )
```

Comment:
Just want to make sure you manually tested the admin interface with this. This secret key is used internally by flask and I don't know at what point it needs to be set.

Reply:
After some travails we confirmed that the key is still available to Flask soon enough.